ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1532,9 +1532,12 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
           getOptions().ageAtAccessTracker.trackAgeAtAccess(serverBlobInfo.getBlobProperties().getCreationTimeInMs());
           blobData = blobAll.getBlobData();
           encryptionKey = blobAll.getBlobEncryptionKey();
-          if (encryptionKey == null) {
-            // set blobInfo only if decryption is not required. If not, maybeProcessCallbackAndComplete() will set appropriate
-            // value
+          if (encryptionKey == null || options.getChunkIdsOnly) {
```

Comment:
this comment is unclear.
this case handles blobs that are not encrypted.
why describe what happens when the blob is encrypted ?
Also the code change doesn't match the commit title

Reply:
An example of a case it doesn't handle is "blob is encrypted, but its a getBlob call without getChunkIdsOnly set". In this case, for example, both `encryptionKey == null` will evaluate to false, and `option.getChunkIdsOnly` will evaluate to false.

Reply:
ok so a better comment might be

`
set blobInfo iff 
1. blob is not encrypted or,
2. getChunkIdsOnly is set since we need blobProperties member of blobInfo.
 
Else, maybeProcessCallbackAndComplete() will set it after decryption.
`